resources/measures/upgrade_hvac_condensing_boiler/measure.rb

```diff
@@ -0,0 +1,251 @@
+
+        # original method: set sp manager to constant 140F
+        # spm.to_SetpointManagerScheduled.get.setSchedule(new_setpoint_sched)
+
+        # create new OA reset setpoint manager with settings: hw temp 140f for air temp <20F, hw temp 120F for air temp >50F
+        oa_reset_sp_mgr = OpenStudio::Model::SetpointManagerOutdoorAirReset.new(model)
+        oa_reset_sp_mgr.setSetpointatOutdoorHighTemperature(OpenStudio.convert(140, 'F', 'C').get)
+        oa_reset_sp_mgr.setOutdoorHighTemperature(OpenStudio.convert(50, 'F', 'C').get)
+        oa_reset_sp_mgr.setSetpointatOutdoorLowTemperature(OpenStudio.convert(180, 'F', 'C').get)
+        oa_reset_sp_mgr.setOutdoorLowTemperature(OpenStudio.convert(20, 'F', 'C').get)
+
+        # add to supply outlet node
+        oa_reset_sp_mgr.addToNode(htg_loop.supplyOutletNode)
+        runner.registerInfo('Added outdoor air reset setpoint manager to hot water loop with 140F for ≤20F and 120F for ≥50F.')
+
+        # autosize boiler supply pump
+        htg_loop.supplyComponents.each do |sup_comp|
+          next unless sup_comp.to_PumpVariableSpeed.is_initialized
+
+          sup_comp.to_PumpVariableSpeed.get
+          # runner.registerInfo('at pump')
+          # pump.autosizeRatedFlowRate
+          # pump.autosizeRatedPowerConsumption
+        end
+
+        boiler.setName('Main Condensing Boiler Thermal Eff 0.95')
+      end
+    end
+
+    # re-autosize all supply fans on airloop
+    model.getAirLoopHVACs.each do |air_loop|
+      air_loop.supplyComponents.each do |sup_comp|
+        if sup_comp.to_FanConstantVolume.is_initialized
+          sup_comp.to_FanConstantVolume.get
+          # const_vol_fan.autosizeMaximumFlowRate
+        elsif sup_comp.to_FanVariableVolume.is_initialized
+          sup_comp.to_FanVariableVolume.get
+          # var_vol_fan.autosizeMaximumFlowRate
+        end
+      end
+    end
+
+    # re-autosize traditional hot water coils
+    hw_coils = model.getCoilHeatingWaters
+    hw_coils.each do |coil|
+      # get hot water coils and reset inlet and outlet temps and autosize
+      coil.setRatedInletWaterTemperature(hw_setpoint_c)
+      coil.setRatedOutletWaterTemperature(hw_return_temp_c)
+      # coil.setRatedInletAirTemperature(12.8) #set to 55F
+      # coil.setRatedOutletAirTemperature(32.2) #set return air temp to 90F
+      # coil.autosizeMaximumWaterFlowRate
+      # coil.autosizeUFactorTimesAreaValue
+      # coil.autosizeRatedCapacity
+    end
+
+    # also check for hot water baseboard coils
+    baseboard_hw_coils = model.getCoilHeatingWaterBaseboards
+    baseboard_hw_coils.each do |coil|
+      # get hot water baseboard coils and  autosize
+      # coil.autosizeMaximumWaterFlowRate
+      # coil.autosizeUFactorTimesAreaValue
+      # coil.autosizeHeatingDesignCapacity
+    end
+
+    # re-autosize zone air terminals and fan coil units to accommodate for new air flow rates
+    thermal_zones = model.getThermalZones
+    thermal_zones.each do |zone|
```

Comment:
Same stuff, don't touch all thermal zones.

Maybe this:

```ruby
coils.each do |c|
   # Some coils tend to report the containing ZoneHVACComponent such as AirLoopHVACUnitarySystem has a containingHVACComponent and not containingZoneHVACComponent
   if c.containingZoneHVACComponent.is_initialized || (c.containingHVACComponent.is_initialized && c.containingHVACComponent.get.to_ZoneHVACComponent.is_initialized) 
        comp = c.containingZoneHVACComponent.is_initialized ? c.containingZoneHVACComponent.get : c.containingHVACComponent.get.to_ZoneHVACComponent.get
         if comp.thermalZone.is_initialized:
              z = comp.thermalZone.get
              # keep on
      end
   end
end
```

Reply:
I think you'd have to grab the zone sizingZone object and adjust the Zone Heating Design Supply Air Temperature